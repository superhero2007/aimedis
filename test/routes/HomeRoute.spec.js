import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import HomeRoute from 'routes/HomeRoute';

describe('routes/HomeRoute', () => {
  it('should render the Home Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/']}>
        <HomeRoute
          exact
          path="/"
          component={() => (<div>HOME</div>)}
          isAuthenticated={false}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });

  it('should render the Home Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/']}>
        <HomeRoute
          exact
          path="/"
          component={() => (<div>HOME</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
