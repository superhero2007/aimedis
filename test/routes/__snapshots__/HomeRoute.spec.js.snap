// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/HomeRoute should render the Home Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">HOME</div>"`;

exports[`routes/HomeRoute should render the Home Page for unauthenticated access 1`] = `"<div data-reactroot=\\"\\">HOME</div>"`;
